### ISSUE TITLE
How do I change to ES6 so far as this is concerned?

### ISSUE THREAD

**Issue by excenter** [AtomLinter/linter-jshint#237]
I've been using this for a while, but where I'm at we use ES6, and have a gulp task to write an ES5 version until there's enough browser support. 
I have 2 parallel requests: 

- A basic explanation on how to permanently change it to ES6 support

- A setting in the preference pane (or the atom config file) that changes the ES version to 6. 

I've read https://github.com/AtomLinter/linter-jshint/issues/202 and it doesn't help, it just shrugs off the setting to somewhere else and doesn't explain where it goes to.

**Comment by Arcanemagus** [comment]
You need to properly configure your `jshint` for this (through the `.jshintrc` file), all this package does is run `jshint` on the open file and report the results.

If you have `jshint` properly configured, and `jshint` from the CLI is reporting different results from what you are seeing inside Atom, then that is a bug that needs to be looked into.

**Comment by excenter** [comment]
where is this file!? Looking through my computer, there are over 40 files named .jshintrc.

**Comment by excenter** [comment]
what words do I put in what file? If I had it properly configured, I wouldn't be here asking questions. 
This package owns whatever version of jshint is throwing the error. 
What changes do I put in which version of jshintrc? in the ` /.atom/packages/linter-jshint/ `  folder has two of them, and adding `"esversion":6` to both of them does nothing.
This hasn't helped at all?

**Comment by Arcanemagus** [comment]
@excenter You should start by reading the Configuration section of the [`jshint` documentation](http://jshint.com/docs/). This will provide a basic idea of how configuring `jshint` works.

In general you place a `.jshintrc` file in the root of your project, this file is read whenever you run the linter on your project. _This_ project, `linter-jshint` is _only_ a translator between the output of running `jshint` on a file, and what the `linter` Atom package is expecting in order to display the issues within the Atom editor.

Neither of the `.jshintrc` files in this packages folders should _ever_ have a effect on your project, as they are only for testing this package's internal functionality.

Hopefully that helps lead you in the right direction, but it isn't the goal of this package to be a tutorial on `jshint` itself (If I was writing that I would tell you to use `eslint` as it is a _much_ better linter in my opinion...), you should familiarize yourself with `jshint` using any of the resources that can be found from a quick Google search.

**Comment by excenter** [comment]
Thank you! @Arcanemagus 
The solution was to make a .jshintrc file at my root developer folder, where the contents are just
```{
  "esversion":6
}```
`/*jshint esversion: 6 */` will throw an error when atom boots up because it's not JSON formatted correctly.

**Comment by BenGitsCode** [comment]
@excenter when you say root developer folder, do you mean your individual project root every time? I'm still getting the W104 error in atom after that.

**Comment by excenter** [comment]
In a folder parent to the working directory (the starred ones)
Developer *
\ Category *
  \Project []
In my Developer folder, I created a file called **.jshintrc** that contained `{ "esversion":6 }` among other options. 
It's really not intuitive

**Comment by BenGitsCode** [comment]
Not intuitive at all, but that fix worked for me. Now I can stop wrapping every goddamn javascript file in that ugly use strict function.

**Comment by BenGitsCode** [comment]
Not intuitive at all, but it worked. At least now I can stop wrapping every javascript file in that hideous function form of use strict.

**Comment by brody182** [comment]
just put /*jshint esversion: 6 */ at the top of your js file

**Comment by revintec** [comment]
@broros Your solution works great, don't know why this is downvoted, I upvoted it.
I don't want change es6 settings globally, but rather only on certain files.

**Comment by 1greenapplee** [comment]
```
{
  "jasmine": true,
  "mocha": true,
  **"esversion":6,**
  "globals": {
    **"esversion": 6,**
    "angular": false,
    "browser": false,
    "inject": false,
    "_": false,
    "driver": false,
    "protractor": false,
    "$": false,
    "$$": false,
    "element": false,
    "by": false,
    "list": false
  }
}
```

This worked for me, without adding "/*jshint esversion: 6 */ " on top of each JS file.

**Comment by IAmNatch** [comment]
@1greenapplee which file did you write this in^

**Comment by 1greenapplee** [comment]
.jshintrc 
Are you using webstorm for this? in this case, you will need to load the config file.

**Comment by IAmNatch** [comment]
I've added esversion and global esversion to the .jshintrc and I'm still getting linted for using const. Any ideas? I really don't want to create a new config file for every project I want to lint properly...

**Comment by 1greenapplee** [comment]
are you using webstorm?(or similar jetbrains product)

**Comment by 1greenapplee** [comment]
If you are using webstorm, or jetbrains products, this is where the config is being set.

![01](https://user-images.githubusercontent.com/16177534/28200202-445c9942-6873-11e7-9a0d-f49e9fec426f.PNG)

**Comment by tkshehan** [comment]
I put this in my User Settings in Visual Studio Code.

`"jshint.options" : {"esversion": 6}`

![capture](https://user-images.githubusercontent.com/36768468/37873606-ad3cb4a8-2fe5-11e8-9369-a48a21132d97.GIF)
